wlauto/workloads/googlephotos/__init__.py

```diff
@@ -0,0 +1,131 @@
+                  description="""
+                  A list of four image files to be pushed to the device.
+                  Absolute file paths may be used but tilde expansion is not supported.
+                  """),
+    ]
+
+    def validate(self):
+        super(Googlephotos, self).validate()
+        self.uiauto_params['package'] = self.package
+        self.uiauto_params['output_dir'] = self.device.working_directory
+        self.uiauto_params['dumpsys_enabled'] = self.dumpsys_enabled
+
+        self._check_image_numbers()
+        self._check_image_extensions()
+        self._check_image_duplicates()
+        self._info_image_used()
+
+    def initialize(self, context):
+        super(Googlephotos, self).initialize(context)
+        for image in self.test_images:
+            if os.path.exists(image):
+                image_path = image
+            else:
+                image_path = context.resolver.get(wlauto.common.resources.File(self, image))
+            self.device.push_file(image_path, self.device.working_directory, timeout=300)
+
+        # Force a re-index of the mediaserver cache to pick up new files
+        self.device.execute('am broadcast -a android.intent.action.MEDIA_MOUNTED -d file:///sdcard')
+
+    def teardown(self, context):
+        super(Googlephotos, self).teardown(context)
+        self.device.execute('am broadcast -a android.intent.action.MEDIA_MOUNTED -d file:///sdcard')
+
+    def finalize(self, context):
+        super(Googlephotos, self).finalize(context)
+
+        for entry in self.device.listdir(self.device.working_directory):
+            if entry in self.test_images:
+                self.device.delete_file(self.device.path.join(self.device.working_directory, entry))
+
+        self.device.execute('am broadcast -a android.intent.action.MEDIA_MOUNTED -d file:///sdcard')
+
+    # -------------------------------------------------------------------------
+    # Internal methods
+    # -------------------------------------------------------------------------
+
+    def _check_image_extensions(self):
+        for image in self.test_images:
+            if not image.endswith(('jpg', 'jpeg', 'png')):
+                raise ValidationError('{} must be a jpeg or png file'.format(image))
+
+    def _check_image_numbers(self):
+        if len(self.test_images) != 4:
+            message = "This workload requires four test images - only {} specified"
+            raise ValidationError(message.format(len(self.test_images)))
+
+    def _check_image_duplicates(self):
+        if len(self.test_images) != len(set(self.test_images)):
+            raise ValidationError('Duplicate image names not allowed')
+
+    def _info_image_used(self):
+        if set(self.test_images) & set(self.default_test_images):
+            self.logger.info('Using default test images')
+        else:
+            self.logger.warning('Using custom test images')
```

Comment:
This is not the right way to do this.

1. A warning implies something has gone wrong; which is not the case here.
2. Someone looking at results is unlikely to go through the log file, so they won't see.

My suggestion is that either the images themselves are saved as part of the results or their names and hashes are (e.g. in a file in the output called "USED_IMAGES").